2-WebApp-graph-user/2-5-HybridFlow/README.md

```diff
@@ -0,0 +1,480 @@
+
+#### Register the service app (TodoListService-aspnetcore-webapi)
+
+  1. Navigate to the [Azure portal](https://portal.azure.com) and select the **Azure Active Directory** service.
+  1. Select the **App Registrations** blade on the left, then select **New registration**.
+  1. In the **Register an application page** that appears, enter your application's registration information:
+     * In the **Name** section, enter a meaningful application name that will be displayed to users of the app, for example `HybridFlow-aspnetcore`.
+  1. Under **Supported account types**, select **Accounts in this organizational directory only**
+  1. Click **Register** to create the application.
+  1. In the app's registration screen, find and note the **Application (client) ID**. You'll need to use this value in your app's configuration files.
+  1. In the app's registration screen, select **Authentication** in the menu.
+     * If you don't have a platform added, select **Add a platform** and select the **Web** option.
+     * In the **Redirect URIs** section, enter the following redirect URIs.
+        * `https://localhost:44321/signin/`
+     * Click on the **Add a platform** button in the **Platform configurations** section of the page
+       * Select the **Single-page application** button and enter `https://localhost:7089/` as the **Redirect URI** and click the **Configure** button
+     * Under the **Implicit grant and hybrid flows** section enable both the **Access tokens** and **ID tokens** options.
+  1. Select **Save** to save your changes.
+  1. In the app's registration screen, select the **Certificates & secrets** blade in the left to open the page where we can generate secrets and upload certificates.
+  1. In the **Client secrets** section, select **New client secret**:
+     * Type a key description (for instance `app secret`),
+     * Select one of the available key durations (**6 months**, **12 months** or **Custom**) as per your security posture.
+     * The generated key value will be displayed when you select the **Add** button. Copy and save the generated value for use in later steps.
+     * You'll need this key later in your code's configuration files. This key value will not be displayed again, and is not retrievable by any other means, so make sure to note it from the Azure portal before navigating to any other screen or blade.
+  1. In the app's registration screen, select the **API permissions** blade in the left to open the page where we add access to the APIs that your application needs.
+     * Select the **Add a permission** button and then,
+     * Ensure that the **Microsoft APIs** tab is selected.
+     * In the *Commonly used Microsoft APIs* section, select **Microsoft Graph**
+     * In the **Delegated permissions** section, select the **User.Read** in the list followed by **Mail.Read**. Use the search box if necessary.
```

Comment:
>Mail.Read [](http://example.com/codeflow?start=100&length=9)

Mail.read requires an email license, can we replace this with some other simple API, like get group memberships ([https://docs.microsoft.com/en-us/graph/api/user-list-memberof](https://docs.microsoft.com/en-us/graph/api/user-list-memberof))